src/options.c

```diff
@@ -283,17 +149,201 @@ find_location_provider(
 	return provider;
 }
 
+/**
+ *
+ * Load options from Elektra, store them in the passed options struct and configure the gamma_methods and location_providers supported in this build..
+ *
+ * @param options A pointer to the options struct where values should be stored.
+ * @param elektra A pointer to the elektra instance.
+ * @param gamma_methods A pointer to the gamma_methods supported in this build of redshift..
+ * @param location_providers A pointer to the location_providers. supported in this build of redshift.
+ * @return 0 if loading was successful, -1 on error
+ */
+int
+options_load_from_elektra(
+        options_t *options,
+        Elektra *elektra,
+        const gamma_method_t *gamma_methods,
+        const location_provider_t *location_providers) {
+    /**
+     * Before using Elektra, there were two sources for configuration: CLI options and config file.
+     * From redshift's point of view there is only one source now, namely Elektra.
+     */
+    // BEGIN Block: Options from parse_command_line_option and parse_config_file_option
+
+    // Programm mode
+    ElektraEnumMode mode = elektraGetMode(elektra);
+    if (mode == ELEKTRA_ENUM_MODE_CONTINUAL) {
```

Comment:
switch so that adding modes gives warnings?

Reply:
Good idea, I refactored into switch.